pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -117,6 +124,33 @@ func New(port uint, runtime string) (Interface, error) {
 	return cadvisorClient, nil
 }
 
+func generateCollectorHttpClient(cAdvisorCollectorClientCertFile, cAdvisorCollectorClientPrivateKeyFile string) (*http.Client, error) {
+	//Enable accessing insecure endpoints. We should be able to access metrics from any endpoint
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
> I'm having trouble parsing out whether you're pulling or pushing

Its pulling metrics, if metrics were being pushed into the system then setting InsecureSkipVerify wouldn't really make any sense.

> But if someone has chosen to set up a secured transport, not verifying that transport defies expectation. It is worse that using http, because it is deceptive.

I don't understand this. If I run a site under https, its not my concern or responsibility if the client is actually verifying the certificates or not. I cannot control how clients are accessing it. Running a site under https gives zero control over what the client is doing and this is not something someone running a service under https would expect.